pkg/api/authn.go

```diff
@@ -127,9 +127,11 @@ func (amw *AuthnMiddleware) basicAuthn(ctlr *Controller, userAc *reqCtx.UserAcce
 			userAc.AddGroups(groups)
 			userAc.SaveOnRequest(request)
 
-			// saved logged session
-			if err := saveUserLoggedSession(cookieStore, response, request, identity, ctlr.Log); err != nil {
-				return false, err
+			// saved logged session only if the request comes from web (has UI session header value)
+			if hasSessionHeader(request) {
```

Comment:
using the current library, no... I can not get a reference for an expired session.
In case of an expired session, the library will return an error (expired session) and a new valid session. losing the ability to get the reference of the expired session and delete it.

Reply:
there are a lot of issues about this:
https://github.com/gorilla/sessions/issues/160

they say how can you delete, but they don't say how to delete expired session, in another issue they state that they are overwritten by new sessions, but that's not true, at least for our case.

Reply:
IMHO let's fix what we can be fixed with this library, and track relacing it with something else (another sessions library? or tokens? we need a discussion) separately.